src/lightning/pytorch/trainer/connectors/accelerator_connector.py

```diff
@@ -505,13 +503,13 @@ def _check_strategy_and_fallback(self) -> None:
 
     def _init_strategy(self) -> None:
         """Instantiate the Strategy given depending on the setting of ``_strategy_flag``."""
+        # The validation of `_strategy_flag` already happened earlier on in the connector
+        assert isinstance(self._strategy_flag, (str, Strategy))
         if isinstance(self._strategy_flag, str):
             self.strategy = StrategyRegistry.get(self._strategy_flag)
-        elif isinstance(self._strategy_flag, Strategy):
+        else:
             # TODO(fabric): remove ignore after merging Fabric and PL strategies
             self.strategy = self._strategy_flag  # type: ignore[assignment]
-        else:
-            raise RuntimeError(f"{self.strategy} is not valid type: {self.strategy}")
```

Comment:
This was reachable, but caused attribute error because self.strategy was not defined yet. 
Now, we validate input earlier just like in Fabric.